Doc/library/string.rst

```diff
@@ -319,12 +319,12 @@ non-empty format specification typically modifies the result.
 The general form of a *standard format specifier* is:
 
 .. productionlist:: format-spec
-   format_spec: [[`fill`]`align`][`sign`]["z"]["#"]["0"][`width`][`grouping_option`]["." `precision`][`type`]
+   format_spec: [[`fill`]`align`][`sign`]["z"]["#"]["0"][`width`][`grouping`]["." `precision`][`type`]
```

Comment:
Maybe it worth also backport other changes in format-spec.  It seems more readable in the 3.14.

Reply:
I'll have a look tomorrow or this evening if I'm not too tired to think about (if you want my thoughts, it'd be to see how it's rendered online as well)

Reply:
Before:
![image](https://github.com/user-attachments/assets/03c67198-4079-47e1-96ed-77321f2034a7)

After:
![image](https://github.com/user-attachments/assets/2fab1992-ad4f-477a-b631-0cd14c5dc672)

Reply:
I personally prefer the first. But I can understand why someone would prefer the second. Maybe we should have done something in Sphinx where we have a max width and auto-wrapping.

Now, do we have other BNFs where we need multiple lines for a single rule? do we usually put `|` at the beginning or the end of the line? (because it could be easier to read if the "b" is aligned with the "G" for instance: all those are 1-char wide)

Reply:
That wrap was introduced by @skirpichev in main, I've just copied it here. I personally do not know if there are examples in other places.

Also note, that for whatever reason main differently renders BNF definitions: main puts only `:` and immediately after the name, 3.13- puts `::=` and aligns it on the longest name. You can check docs preview from main PR to spot the difference.

Regarding the placing of `|`: in multiline expressions I prefer putting operators in the beginning of new lines as this is how it is done in many math textbooks. However, I agree on aligning, it would be better if `|` is under `=` (so `b` and `G` are in the same column), but I have no clue is it even possible to do so.

Reply:
It seems that other tables put `|` at the beginning. And on main it looks the same. Let's not change it too much.

> Also note, that for whatever reason main differently renders BNF definitions

That's because we changed something in Sphinx (or was it a custom extension, anyway we changed it only on main).

Reply:
IOW: Use your suggestion where you break the list into two.

Reply:
Alright, that change is already in this PR. Resolving this.